types/V1.mli

```diff
@@ -19,7 +19,27 @@
 (** MirageOS Signatures.
 
     This module defines the basic signatures that functor parameters
-    should implement in MirageOS to be portable. *)
+    should implement in MirageOS to be portable.
+
+    {1 General conventions}
+
+    {ul
+    {- errors which application programmers are expected to handle
+       (e.g. connection refused or end of file) are encoded as result types
+       where [`Ok x] means the operation was succesful and returns [x] and
+       where [`Error e] means the operation has failed with error [e]. }
+    {- errors which represent programming errors such as assertion failures
+       or illegal arguments are encoded as exceptions and the application
+       is expected to output diagnostics and abort. }
```

Comment:
This is a little bit unclear. I'd say that either the application tries to catch the exceptions and tries to recover (which may not be a good idea) or you let the whole thing crash and it should be the runtime's system duty to output diagnostics and abort.